agenta-backend/agenta_backend/services/app_manager.py

```diff
@@ -157,7 +157,7 @@ async def terminate_and_remove_app_variant(
 
             image = app_variant_db.base.image
             logger.debug("is_last_variant_for_image {image}")
-            if image:
+            if image and image.deletable:
```

Comment:
I don't think this logic is correct. We still need to stop_and_delete the service even when the image is deletable. The only things that we should skip are:
- removing the image (line 166)
- removing the image db (line 170)

Or am I mistaken?

Reply:
You said this:
> We need to make sure not to delete either the ImageDB object nor its deployment when we have deletable=False. 

The only resources that would be deleted for apps created with a template are lines 168 and 172. But then, if we decide to remove deployments, we'd have to remove line 172 (base) which also has a dependency on line 162 (deployment), which makes me also remove line 171.